test/test_dynproc.py

```diff
@@ -164,14 +164,29 @@ def testJoin(self):
         size = MPI.COMM_WORLD.Get_size()
         rank = MPI.COMM_WORLD.Get_rank()
         server = client = address = None
-        # crate server/client sockets
+        host = socket.gethostname()
+        addresses = socket.getaddrinfo(host, None)
+        address_families = [ a[0] for a in addresses if a[1]==socket.SOCK_STREAM ]
+        # if both INET and INET6 are available, don't assume the order
+        # is the same on both server and client. Select INET if available.
+        if socket.AF_INET in address_families:
+            socket_family = socket.AF_INET
+        elif socket.AF_INET6 in address_families:
+            socket_family = socket.AF_INET6
+        elif socket.AF_UNIX in address_families:
+            socket_family = socket.AF_UNIX
+        elif address_families:
+            # allow for an undocumented family
+            socket_family = address_families[0]
+        else:
+            raise ConnectionError("internet socket not found")
+        # create server/client sockets
```

Comment:
Shouldn't we check that `socket_family` is the same in different processes?

Reply:
That was the motivation for explicitly selecting AN_INET as first choice, so that all processes would invoke the same family.  This assumes that the set of families is the same on all processes, but I didn't assume that the order returned by getaddrinfo() would necessarily be same.

If it's a situation where server has AN_INET (with or without AN_INET6), while client has AN_INET6 only, then it will be a problem.  I think it's not constructive (i.e. too complex) to then require server to switch to AN_INET6.  But checking socket_family across processes is also complex, requiring MPI transfer of the value.  The only step to take if it is done would be to raise an error.  But I imagine an error (or at least a timeout) will be raised anyway if the families don't match.

It seems to me the complexity of catching that scenario would make the test over-engineered.  It would be a strange configuration for client and server to be running on different network families. When it does happen then the test would fail anyway.  Is the scenario likely to occur in real installations?

Reply:
> It seems to me the complexity of catching that scenario would make the test over-engineered

You know, buddy, I'm so sick of dealing with MPI failures that I admittedly overreact.

> Is the scenario likely to occur in real installations?

Probably not, that would sound like be a crazy setup. But never underestimate the ability of people to mess things up.

Let's keep it as it is for now.